Tests/SentryTests/Integrations/SessionReplay/SentrySessionReplayIntegrationTests.swift

```diff
@@ -68,67 +77,90 @@ class SentrySessionReplayIntegrationTests: XCTestCase {
-        expect(SentryGlobalEventProcessor.shared().processors.count) == 1
+        XCTAssertEqual(SentrySDK.currentHub().trimmedInstalledIntegrationNames().count,1)
+        XCTAssertEqual(SentryGlobalEventProcessor.shared().processors.count,1)
     }
     
     func testWaitForNotificationWithNoWindow() {
         uiApplication.windowsMock = nil
         startSDK(sessionSampleRate: 1, errorSampleRate: 0)
         
-        guard let sut = SentrySDK.currentHub().installedIntegrations().first as? SentrySessionReplayIntegration else {
-            fail("Did not installed replay integration")
-            return
-        }
+        guard let sut = getSut() else { return }
         
-        expect(Dynamic(sut).sessionReplay.asObject) == nil
+        XCTAssertNil(sut.sessionReplay)
         uiApplication.windowsMock = [UIWindow()]
         NotificationCenter.default.post(name: UIScene.didActivateNotification, object: nil)
-        expect(Dynamic(sut).sessionReplay.asObject) != nil
+        XCTAssertNotNil(sut.sessionReplay)
     }
     
     func testPauseAndResumeForApplicationStateChange() {
         startSDK(sessionSampleRate: 1, errorSampleRate: 0)
         
-        guard let sut = SentrySDK.currentHub().installedIntegrations().first as? SentrySessionReplayIntegration else {
-            XCTFail("Did not find Session Replay Integration")
-            return
-        }
+        guard let sut = getSut() else { return }
         
         NotificationCenter.default.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
         XCTAssertFalse(Dynamic(sut.sessionReplay).isRunning.asBool ?? true)
         NotificationCenter.default.post(name: UIApplication.willEnterForegroundNotification, object: nil)
         XCTAssertTrue(Dynamic(sut.sessionReplay).isRunning.asBool ?? false)
     }
+    
+    func testStopReplayAtEndOfSession() {
+        startSDK(sessionSampleRate: 1, errorSampleRate: 0)
+        
+        guard let sut = getSut() else { return }
+        XCTAssertNotNil(sut.sessionReplay)
+        SentrySDK.currentHub().endSession()
+        XCTAssertNil(sut.sessionReplay)
+    }
+ 
+    func testStartFullSessionForError() {
+        startSDK(sessionSampleRate: 0, errorSampleRate: 1)
+        guard let sut = getSut() else { return }
+        
+        XCTAssertFalse(Dynamic(sut.sessionReplay).isFullSession.asBool ?? true)
+        SentrySDK.capture(error: NSError(domain: "", code: 1))
+        XCTAssertTrue(Dynamic(sut.sessionReplay).isFullSession.asBool ?? false)
+    }
+    
+    func testRestartReplayWithNewSession() {
+        startSDK(sessionSampleRate: 1, errorSampleRate: 0)
+        
+        guard let sut = getSut() else { return }
+        XCTAssertNotNil(sut.sessionReplay)
+        SentrySDK.currentHub().endSession()
+        XCTAssertNil(sut.sessionReplay)
+        SentrySDK.currentHub().startSession()
+        XCTAssertNotNil(sut.sessionReplay)
```

Comment:
`l`: Does it make sense to do some asserts here to validate the `sessionReplay`?

Reply:
I don't think so. if it was initialized is all good, SR behaviour has its own tests.